os/kernel/debug/sysdbg.c

```diff
@@ -628,8 +803,55 @@ void save_task_scheduling_status(struct tcb_s *tcb)
 #endif
 		sysdbg_struct->sched[index].pid = tcb->pid;
 		sysdbg_struct->sched[index].sched_priority = tcb->sched_priority;
-		sysdbg_struct->sched[index].ptcb = tcb;
+		sysdbg_struct->sched[index].context_switch_reason = CONTEXT_DEFAULT;
+	}
+
+	/* Assign context switch reason based on task states */
+	switch (tcb->task_state) {
```

Comment:
This is at out of tcb checking. But you assume the tcb could be null by line 800. Need to check again.